packages/client/src/__tests__/dataTriggers.test.ts

```diff
@@ -674,4 +675,177 @@ describe("Cloud functions -> Data triggers ->", () => {
       }
     );
   });
+  describe("createCustomerStats", () => {
+    testWithEmulator(
+      "should update the customer doc with bookingStats with respect to customer's bookedSlots",
+      async () => {
+        const { organization } = await setUpOrganization();
+        const dateThisMonth = DateTime.now().toISODate();
+        const dateNextMonth = DateTime.now().plus({ month: 1 }).toISODate();
+        const slotThisMonthIce = {
+          ...baseSlot,
+          date: dateThisMonth,
+          type: SlotType.Ice,
+        };
+        const slotNextMonthIce = {
+          ...baseSlot,
+          date: dateNextMonth,
+          type: SlotType.Ice,
+        };
+        const slotThisMonthOffIce = {
+          ...baseSlot,
+          date: dateThisMonth,
+          type: SlotType.OffIce,
+        };
+        const slotNextMonthOffIce = {
+          ...baseSlot,
+          date: dateNextMonth,
+          type: SlotType.OffIce,
+        };
+        // Create customer and slot
+        await Promise.all([
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateThisMonth}-9`))
+            .set(slotThisMonthIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateNextMonth}-9`))
+            .set(slotNextMonthIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateThisMonth}-10`))
+            .set(slotThisMonthOffIce),
+          adminDb
+            .doc(getSlotDocPath(organization, `${dateNextMonth}-10`))
+            .set(slotNextMonthOffIce),
+          adminDb.doc(getCustomerDocPath(organization, saul.id)).set(saul),
+        ]);
+
+        await Promise.all([
+          await waitFor(async () => {
+            const snap = await adminDb
+              .doc(getCustomerDocPath(organization, saul.id))
+              .get();
+            expect(snap.data()).toEqual(saul);
+          }),
+        ]);
```

Comment:
I understand the additional check (just-to-make-sure), but when we store saul into customers collection (in the previous `Promise.all` block), the saul will be there (we need to do this check only when we depend on a data trigger being ran in the background and want to verify it had completed)